paperqa/agents/search.py

```diff
@@ -270,6 +278,16 @@ async def mark_failed_document(self, path: str | os.PathLike) -> None:
         (await self.index_files)[str(path)] = FAILED_DOCUMENT_ADD_ID
         self.changed = True
 
+    async def release_lock(self) -> None:
```

Comment:
So this feels like a hack that we have added due to a lack of understanding the locks

Can we understand why we are encountering deadlocks, and do a more precise fix